app/models/miq_queue.rb

```diff
@@ -116,6 +116,11 @@ def self.put(options)
 
     options[:args] = [options[:args]] if options[:args] && !options[:args].kind_of?(Array)
 
+    if !Rails.env.production? && options[:args] &&
+       (arg = options[:args].detect { |a| a.kind_of?(ActiveRecord::Base) && ! a.new_record? })
+      raise "MiqQueue.put(:args =>[]) does not support #{arg.class.name} objects"
```

Comment:
This doesn't help people fix the code though... should we suggest you provide class and id so the resulting method can lookup the record instead of us serializing it on the queue?